R/net.mod.cumulative_edge_list.R

```diff
@@ -0,0 +1,37 @@
+#' @title Cumulative Edgelist: netsim Module
+#'
+#' @description This function generate the cumulative edgelists for each network
+#'              in the model.
+#'
+#' @param dat Master list object containing a \code{networkDynamic} object and
+#'        other initialization information passed from \code{\link{netsim}}.
+#' @param at Current time step.
+#'
+#' @return the updated dat Master list object.
+#'
+#' @section Optional Module:
+#' This module is not included by default
+#'
+#' @section Truncation:
+#' To avoid storing a cumulative edgelist to long, the "truncate.el_cuml"
+#' control value defines a number of steps after which an edge that is no longer
+#' active is truncated out of the cummulative edgelist. (see \code{control.net})
+#'
+#' @section Accessing the Cumulative Edgelists:
+#' This module only computes and store the cumulative edgelists. They can then
+#' be accessed using \code{\link{get_cumulative_edgelist}}
+#' \code{\link{get_cumulative_edgelists_df}}
+#
+#' @seealso \code{\link{get_cumulative_edgelist}}
+#' \code{\link{get_cumulative_edgelists_df}} \code{\link{netsim}}
+#'
+#' @export
+#' @keywords netMod internal
+#'
+cumulative_edgelist.net <- function(dat, at) {
```

Comment:
I don't think this should be a standalone, top-level module. It is just two functions (or sets of functions) to update and then get the cuml edgelist that would be called within two places: nwupdate.net (for the updating) and any module that needs that cuml el data (e.g., a contact tracing module).

Reply:
Because of that, I would suggest setting the control.net setting `truncate.el.cuml` default to 0, which means, don't store this data at all. If `truncate.el.cuml >= 1`, then it gets run. It doesn't need to get saved to the output netsim object, but it could potentially if we specified `el_cuml` in `save.other` in `control.net`

Reply:
I agree with all this. I will make the changes

Reply:
New thoughts on that while making it:

I think we might want another control like `cumulative.edgelist` (logical) to ask whether or not to compute the cumulative edgelist and then `truncate.el.cuml`. 

The reasoning is that by default we probably don't want to calculate the `el_cuml` but activating that with the truncation control feels weird. Also, sometimes we might want to calculate the `el_cuml` with `truncate.el.cuml == 0` to keep only the active partnerships but still know their start step. 

What do you think? 

Also, if we go this route, if `truncate.el.cuml` is `NULL`, should we consider that it's `0` or `Inf` (current behavior)